File: tesis_movil/lib/models/medicamento.dart
class Medicamento {
  final int idMedicamento;
  final String nombre;
  final String? principioActivo;
  final String? concentracion;
  final String? presentacion;
  final int cantidadDisponible;
  final int stockMinimo;
  final String? fechaVencimiento; // <--- Agregamos este campo nuevamente

  Medicamento({
    required this.idMedicamento,
    required this.nombre,
    this.principioActivo,
    this.concentracion,
    this.presentacion,
    required this.cantidadDisponible,
    required this.stockMinimo,
    this.fechaVencimiento, // <--- Agregamos al constructor
  });

  factory Medicamento.fromJson(Map<String, dynamic> json) {
    return Medicamento(
      idMedicamento: json['id_medicamento'] ?? 0,
      nombre: json['nombre'] ?? 'Sin Nombre',
      principioActivo: json['principio_activo'],
      concentracion: json['concentracion'],
      presentacion: json['presentacion'],
      cantidadDisponible: json['cantidad_disponible'] ?? 0,
      stockMinimo: json['stock_minimo'] ?? 10,
      fechaVencimiento: json['fecha_vencimiento'], // <--- Leemos del JSON
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_medicamento': idMedicamento,
      'nombre': nombre,
      'principio_activo': principioActivo,
      'concentracion': concentracion,
      'presentacion': presentacion,
      'cantidad_disponible': cantidadDisponible,
      'stock_minimo': stockMinimo,
      'fecha_vencimiento': fechaVencimiento, // <--- Enviamos al backend
    };
  }
}

File: tesis-api/models/Medicamento.js
module.exports = (sequelize, DataTypes) => {
    const Medicamento = sequelize.define('Medicamento', {
        id_medicamento: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true
        },
        nombre: {
            type: DataTypes.STRING,
            allowNull: false
        },
        principio_activo: {
            type: DataTypes.STRING
        },
        concentracion: { // Campo separado como pediste
            type: DataTypes.STRING
        },
        presentacion: {
            type: DataTypes.STRING
        },
        cantidad_disponible: {
            type: DataTypes.INTEGER,
            defaultValue: 0
        },
        stock_minimo: {
            type: DataTypes.INTEGER,
            defaultValue: 10
        },
        fecha_vencimiento: {
            type: DataTypes.DATEONLY // DATEONLY guarda solo YYYY-MM-DD
        }
    }, {
        tableName: 'Medicamentos',
        timestamps: false,
        indexes: [
            {
                unique: true,
                fields: ['nombre', 'concentracion', 'fecha_vencimiento']
            }
        ]
    });

    return Medicamento;
};